semantic/poly_types.go

```diff
@@ -785,13 +785,28 @@ func (l ObjectKind) unifyKind(kinds map[Tvar]Kind, k Kind) (Kind, Substitution,
 func (k ObjectKind) resolveType(kinds map[Tvar]Kind) (Type, error) {
 	properties := make(map[string]Type, len(k.properties))
 	for l, ft := range k.properties {
-		if _, ok := ft.(invalid); !ok {
-			t, err := ft.resolveType(kinds)
-			if err != nil {
-				return nil, err
+		if _, ok := ft.(invalid); ok {
+			continue
+		}
+
+		t, err := func() (Type, error) {
+			switch ft := ft.(type) {
+			case Tvar:
+				// Check if this can be resolved.
+				if _, ok := kinds[ft]; !ok {
```

Comment:
If this property has not been assigned a type, shouldn't it be associated with a NullableKind in the `kinds` map? When does `!ok` evaluate to true?

Reply:
Whoops. This was added as my first attempt before using constraints. It didn't work. I'll revert this section.